cmd/upcloud-csi-plugin/main.go

```diff
@@ -3,82 +3,136 @@ package main
+
+	envUpcloudUsername string = "UPCLOUD_USERNAME"
+	envUpcloudPassword string = "UPCLOUD_PASSWORD"
+	envStorageLabels   string = "STORAGE_LABELS"
+)
+
 func main() {
 	flagSet := pflag.NewFlagSet("default", pflag.ContinueOnError)
-
 	var (
-		endpoint     = flagSet.String("endpoint", "unix:///var/lib/kubelet/plugins/"+driver.DefaultDriverName+"/csi.sock", "CSI endpoint")
-		nodeHost     = flagSet.String("nodehost", "", "Node hostname to determine node's zone and UUID")
+		endpoint     = flagSet.String("endpoint", driver.DefaultEndpoint, "CSI endpoint")
+		nodeHost     = flagSet.String("nodehost", "", "Node's hostname. This should match server's `hostname` in the hub.upcloud.com.")
+		zone         = flagSet.String("zone", "", "The zone in which the driver will be hosted, e.g. de-fra1. Defaults to `nodeHost` zone.")
 		username     = flagSet.String("username", "", "UpCloud username")
 		password     = flagSet.String("password", "", "UpCloud password")
 		driverName   = flagSet.String("driver-name", driver.DefaultDriverName, "Name for the driver")
 		address      = flagSet.String("address", driver.DefaultAddress, "Address to serve on")
 		version      = flagSet.Bool("version", false, "Print the version and exit.")
 		isController = flagSet.Bool("is_controller", true, "Run driver with controller included")
-		logLevel     = flagSet.String("log-level", "warning", "Loggin level: panic, fatal, error, warn, warning, info, debug or trace")
+		logLevel     = flagSet.String("log-level", "info", "Loggin level: panic, fatal, error, warn, warning, info, debug or trace")
+		labels       = flagSet.StringSlice("label", nil, "Apply default labels to all storage devices created by CSI driver, e.g. --label=color=green --label=size=xl")
 	)
+	if err := flagSet.Parse(os.Args[1:]); err != nil {
+		log.Fatal(err)
+	}
 
-	err := flagSet.Parse(os.Args[1:])
-	if err != nil {
-		log.Fatalln(err)
+	if *version {
+		printVersionAndExit()
 	}
 
-	if *username == "" {
-		*username = os.Getenv("UPCLOUD_USERNAME")
+	if *nodeHost == "" {
+		log.Fatalf("nodehost missing")
 	}
-	if *password == "" {
-		*password = os.Getenv("UPCLOUD_PASSWORD")
+	svc := service.New(newUpcloudClient(*username, *password))
+	logger := newLogger(*logLevel)
+	options := driver.Options{
+		DriverName:    *driverName,
+		Endpoint:      newEndpoint(*endpoint),
+		Address:       newAddress(*address),
+		NodeHost:      *nodeHost,
+		Zone:          *zone,
+		IsController:  *isController,
+		StorageLabels: newLabels(*labels),
+	}
+	drv, err := driver.NewDriver(svc, options, logger, nil)
+	if err != nil {
+		logger.Fatal(err)
 	}
 
-	if *version {
-		fmt.Printf("%s - %s (%s)\n", driver.GetVersion(), driver.GetCommit(), driver.GetTreeState()) //nolint: forbidigo // allow printing to console
-		os.Exit(0)
+	if err := drv.Run(driver.NewUpcloudHealthChecker(svc)); err != nil {
+		log.Fatalln(err)
 	}
+}
 
-	if *nodeHost == "" {
-		log.Fatalln("nodehost missing")
+func printVersionAndExit() {
```

Comment:
We should align how our UKS components report version info
CAPU and CCM reports it in a similar way, should we utilise that? Not a critical one, just a side comment

Reply:
Sounds good. I'll create separate issue for version reporting :+1: